jms/src/test/scala/akka/stream/alpakka/jms/scaladsl/JmsAckConnectorsSpec.scala

```diff
@@ -0,0 +1,371 @@
+        .map(env => (env, env.message.asInstanceOf[TextMessage].getText))
+        .map { case (env, text) => env.acknowledge(); text }
+        .runWith(Sink.seq)
+
+      result.futureValue should contain theSameElementsAs in
+    }
+
+    "disconnection should fail the stage" in withServer() { ctx =>
+      val connectionFactory = new ActiveMQConnectionFactory(ctx.url)
+      val result = JmsSource
+        .ackSource(
+          JmsSourceSettings(connectionFactory)
+            .withSessionCount(5)
+            .withBufferSize(0)
+            .withQueue("test")
+        )
+        .runWith(Sink.seq)
+      Thread.sleep(500)
+      ctx.broker.stop()
+      result.failed.futureValue shouldBe an[JMSException]
+    }
+
+    "publish and consume elements through a topic " in withServer() { ctx =>
+      import system.dispatcher
+
+      val connectionFactory = new ActiveMQConnectionFactory(ctx.url)
+
+      //#create-topic-sink
+      val jmsTopicSink: Sink[String, NotUsed] = JmsSink.textSink(
+        JmsSinkSettings(connectionFactory).withTopic("topic")
+      )
+      //#create-topic-sink
+      val jmsTopicSink2: Sink[String, NotUsed] = JmsSink.textSink(
+        JmsSinkSettings(connectionFactory).withTopic("topic")
+      )
+
+      val in = List("a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k")
+      val inNumbers = (1 to 10).map(_.toString)
+
+      //#create-topic-source
+      val jmsTopicSource: Source[AckEnvelope, KillSwitch] = JmsSource.ackSource(
+        JmsSourceSettings(connectionFactory).withSessionCount(1).withBufferSize(0).withTopic("topic")
+      )
+      //#create-topic-source
+      val jmsSource2: Source[AckEnvelope, KillSwitch] = JmsSource.ackSource(
+        JmsSourceSettings(connectionFactory).withSessionCount(1).withBufferSize(0).withTopic("topic")
+      )
+
+      val expectedSize = in.size + inNumbers.size
+      //#run-topic-source
+      val result1 = jmsTopicSource
+        .take(expectedSize)
+        .map(env => (env, env.message.asInstanceOf[TextMessage].getText))
+        .map { case (env, text) => env.acknowledge(); text }
+        .runWith(Sink.seq)
+        .map(_.sorted)
+      val result2 = jmsSource2
+        .take(expectedSize)
+        .map(env => (env, env.message.asInstanceOf[TextMessage].getText))
+        .map { case (env, text) => env.acknowledge(); text }
+        .runWith(Sink.seq)
+        .map(_.sorted)
+      //#run-topic-source
+
+      //We wait a little to be sure that the source is connected
+      Thread.sleep(500)
+
+      //#run-topic-sink
+      Source(in).runWith(jmsTopicSink)
+      //#run-topic-sink
+      Source(inNumbers).runWith(jmsTopicSink2)
+
+      val expectedList: List[String] = in ++ inNumbers
+      result1.futureValue should contain theSameElementsAs expectedList
+      result2.futureValue should contain theSameElementsAs expectedList
+    }
+
+    "ensure no message loss when stopping a stream" in withServer() { ctx =>
+      val url: String = ctx.url
```

Comment:
only used in the row below, so no need to put in an explicitly typed val here

Reply:
Yup, this was copied from `JmsConnectorsSpec`. I'll clean it all up (including the old code).

Reply:
While cleaning up, I also see we don't even have to create the val. We can use `ctx.url` directly. There is just one use anyway. Change across the board?